personal_finances/transaction/definition.py

```diff
@@ -1,10 +1,38 @@
-from typing import TypedDict
+from typing import Union
+from typing_extensions import TypedDict
 from datetime import datetime
+from pydantic import TypeAdapter
 
 
 class SimpleTransaction(TypedDict):
     transactionId: str
     datetime: datetime
-    amount: float
+    amount: Union[float, int]
     referenceText: str
     bankTransactionCode: str
+
+
+SimpleTransactionAdapter = TypeAdapter(SimpleTransaction)
+
+
+def _validate_transaction(transaction: SimpleTransaction) -> None:
+    SimpleTransactionAdapter.validate_python(transaction)
+
+
+def simple_transaction_builder(
+    transactionId: str,
+    datetime: datetime,
+    amount: Union[float, int],
+    referenceText: str,
+    bankTransactionCode: str,
+) -> SimpleTransaction:
+    transaction_dict = SimpleTransaction(
+        transactionId=transactionId,
+        datetime=datetime,
+        amount=amount,
+        referenceText=referenceText,
+        bankTransactionCode=bankTransactionCode,
+    )
+    _validate_transaction(transaction_dict)
+
+    return transaction_dict
```

Comment:
Are we keeping these changes here? I suggest creating  another issue / PR for these. It's better that whenever we introduce this builder, the whole codebase stops instantiating `SimpleTransaction` and starts calling this builder.

Reply:
Agree. I have created an issue for that #26